### PR TITLE
Adding custom User-Agent to client

### PR DESCRIPTION
**Pull request by eadgbear**
## 🗣 Description

* Adding a custom user-agent to spicepy to improve the identification of the client when making requests.
* Changing header from `x-spice-user-agent` to standard `User-Agent`

## 🤔 Concerns

### CODE REVIEW

File: spicepy/_http.py [hunk 1]
```diff
@@ -30,7 +30,8 @@ def __init__(self, base_url: str, headers: Dict[str, str]) -> None:
         self.session = self._create_session(headers)
 
         # set the x-spice-user-agent header
```
Comment (sgrebnov) [inline spicepy/_http.py]:
```suggestion
        # set the user-agent header
```

Comment (sgrebnov) [reply spicepy/_http.py]: It seems we always use `SPICE_USER_AGENT` in this case - when it is possible?  Can there be a case that there will always be default user-agent even if custom was provided?

File: spicepy/_http.py [hunk 2]
```diff
@@ -29,8 +29,9 @@ class HttpRequests:
     def __init__(self, base_url: str, headers: Dict[str, str]) -> None:
         self.session = self._create_session(headers)
 
-        # set the x-spice-user-agent header
-        self.session.headers["X-Spice-User-Agent"] = SPICE_USER_AGENT
+        # set the user-agent header
+        if "user-agent" not in self.session.headers:
+            self.session.headers["user-agent"] = SPICE_USER_AGENT
```
Comment (sgrebnov) [inline spicepy/_http.py:34]: It seems we always rely on default user agent only here. Can there be a case when user agent is provided but we fallback on default one here?

Comment (eadgbear) [reply spicepy/_http.py:34]: We rely on it here when the `Client` doesn't set the header already. `_client.py` line 159

**Review by sgrebnov — APPROVED** [review]

**Pull request merged**